pkg/registry/populator.go

```diff
@@ -379,18 +380,43 @@ func (i *DirectoryPopulator) loadOperatorBundle(manifest PackageManifest, bundle
 	return nil
 }
 
+type bundleVersion struct {
+	name    string
+	version semver.Version
+
+	// Keep track of the number of times we visit each version so we can tell if a head is contested
+	count int
+}
+
+// compare returns a value less than one if the receiver arg is less smaller the given version, greater than one if it is larger, and zero if they are equal.
+// This comparison follows typical semver precedence rules, with one addition: a simple ordering is defined for build metadata (+<build-metadata>) corresponding to whole numbers such that they are in ascending order; e.g. 1.0.0 < 1.0.0+1 < 1.0.0+2 ...
+func (b bundleVersion) compare(v bundleVersion) int {
+	if c := b.version.Compare(v.version); c != 0 {
+		return c
+	}
+
+	// Attempt to compare based on build metadata
+	return buildInt(b.version).Cmp(buildInt(v.version))
+}
+
+func buildInt(v semver.Version) *big.Int {
+	build := big.NewInt(0)
+	if len(v.Build) != 1 {
+		// Can only be an integer build-meta with a single element, otherwise treat as zero
+		return build
+	}
+	if b, _ := build.SetString(v.Build[0], 10); b != nil {
```

Comment:
buildIDs don't have to be integers so if i'm understanding this correctly, this seems likely to fail.

Reply:
yup -- the intent is to treat everything outside the domain (positive integers) as zero